parquet-hadoop/src/test/java/org/apache/parquet/crypto/TestPropertiesDrivenEncryption.java

```diff
@@ -0,0 +1,507 @@
+        + "required boolean boolean_field; "
+        + "required int32 int32_field; "
+        + "required float float_field; "
+        + "required double double_field; "
+        + "} ");
+
+    SimpleGroupFactory f = new SimpleGroupFactory(schema);
+    for (Map.Entry<EncryptionConfiguration, Configuration> encryptionConfigurationEntry : encryptionPropertiesMap.entrySet()) {
+      KeyToolkit.removeCacheEntriesForAllTokens();
+      EncryptionConfiguration encryptionConfiguration = encryptionConfigurationEntry.getKey();
+      Configuration conf = encryptionConfigurationEntry.getValue();
+
+      String suffix = (EncryptionConfiguration.NO_ENCRYPTION == encryptionConfiguration) ? ".parquet" : ".parquet.encrypted";
+      Path file = new Path(root, encryptionConfiguration.toString() + suffix);
+      LOG.info("\nWrite " + file.toString());
+
+      FileEncryptionProperties fileEncryptionProperties = null;
+      if (null == conf) {
+        conf = new Configuration();
+      } else {
+        EncryptionPropertiesFactory cryptoFactory = EncryptionPropertiesFactory.loadFactory(conf);
+        fileEncryptionProperties = cryptoFactory.getFileEncryptionProperties(conf, file, null);
+      }
+      ParquetWriter<Group> writer = ExampleParquetWriter.builder(file)
+        .withConf(conf)
+        .withWriteMode(OVERWRITE)
+        .withType(schema)
+        .withEncryption(fileEncryptionProperties)
+        .build();
+      for (int i = 0; i < 100; i++) {
+        boolean expect = false;
+        if ((i % 2) == 0)
+          expect = true;
+        float float_val = (float) i * 1.1f;
+        double double_val = (i * 1.1111111);
+
+        writer.write(
+          f.newGroup()
+            .append("boolean_field", expect)
+            .append("int32_field", i)
+            .append("float_field", float_val)
+            .append("double_field", double_val));
+
+      }
+      writer.close();
+    }
+  }
+
+  private void testReadEncryptedParquetFiles(Path root, Map<DecryptionConfiguration, Configuration> decryptionPropertiesMap) throws IOException {
+    for (Map.Entry<DecryptionConfiguration, Configuration> decryptionConfigurationEntry : decryptionPropertiesMap.entrySet()) {
+      DecryptionConfiguration decryptionConfiguration = decryptionConfigurationEntry.getKey();
+      LOG.info("\n\n");
+      LOG.info("==> Decryption configuration {}\n", decryptionConfiguration);
+
+      File folder = new File(root.toString());
+      File[] listOfFiles = folder.listFiles();
+
+      for (int fileNum = 0; fileNum < listOfFiles.length; fileNum++) {
+        KeyToolkit.removeCacheEntriesForAllTokens();
+        Path file = new Path(listOfFiles[fileNum].getAbsolutePath());
+        if (!file.getName().endsWith(".parquet.encrypted") && !file.getName().endsWith(".parquet")) { // Skip non-parquet files
+          continue;
+        }
+        EncryptionConfiguration encryptionConfiguration = getEncryptionConfigurationFromFilename(file.getName());
```

Comment:
I think, it would be more clean and less error prone if we would iterate on the Encryption configurations and expect the related file to be there instead of skipping all the files that do not match. BTW, there should be no files in that temporary directory that was not created by this test.